internal/provider/harness_docker_resource_test.go

```diff
@@ -0,0 +1,189 @@
+				Config: `
+provider "imagetest" {
+  harnesses = {
+    docker = {
+      envs = {
+        foo = "foo"
+        baz = "override"
+      }
+    }
+  }
+}
+
+data "imagetest_inventory" "this" {}
+
+resource "imagetest_harness_docker" "test" {
+  name = "test"
+  inventory = data.imagetest_inventory.this
+  envs = {
+    bar = "bar"
+    baz = "baz"
+  }
+}
+
+resource "imagetest_feature" "test" {
+  name = "Simple Docker based test"
+  description = "Test that we can spin up a container and run some steps with environment variables"
+  harness = imagetest_harness_docker.test
+  steps = [
+    {
+      name = "Echo"
+      cmd = "echo $foo $bar $baz | diff - <(echo foo bar baz) > /dev/null"
+    },
+  ]
+}
+        `,
+				Check: resource.ComposeAggregateTestCheckFunc(),
+			},
+		},
+		"with working directory": {
+			{
+				ExpectNonEmptyPlan: true,
+				Config: `
+data "imagetest_inventory" "this" {}
+
+resource "imagetest_harness_docker" "test" {
+  name = "test"
+  inventory = data.imagetest_inventory.this
+}
+
+resource "imagetest_feature" "test" {
+  name = "Simple Docker based test"
+  description = "Test that we can spin up a container and run some steps with a working directory"
+  harness = imagetest_harness_docker.test
+  steps = [
+    {
+      workdir = "/tmp"
+      name = "Echo"
+      cmd = "echo test >> .testfile"
+    },
+    {
+      workdir = "/tmp"
+      name = "Cat"
+      cmd = "cat .testfile"
+    }
+  ]
+}
+        `,
+				Check: resource.ComposeAggregateTestCheckFunc(),
+			},
+		},
+		"with volumes configuration": {
+			{
+				ExpectNonEmptyPlan: true,
+				Config: `
+data "imagetest_inventory" "this" {}
+
+resource "imagetest_container_volume" "volume" {
+  name = "volume-test"
+  inventory = data.imagetest_inventory.this
+}
+
+resource "imagetest_harness_docker" "test" {
+  name = "test"
+  inventory = data.imagetest_inventory.this
+  volumes = [
+    {
+      source = imagetest_container_volume.volume
+      destination = "/volume"
+    }
+  ]
+}
+
+resource "imagetest_feature" "test" {
+  name = "Simple Docker based test"
+  description = "Test that we can spin up a container and run some steps with a volume working directory"
+  harness = imagetest_harness_docker.test
+  steps = [
+    {
+      workdir = "/volume"
+      name = "Echo"
+      cmd = "echo test >> .testfile"
+    },
+    {
+      workdir = "/volume"
+      name = "Cat"
+      cmd = "cat .testfile"
+    },
+  ]
+}
+        `,
+				Check: resource.ComposeAggregateTestCheckFunc(),
+			},
+		},
+		"docker works": {
+			{
+				ExpectNonEmptyPlan: true,
+				Config: `
+data "imagetest_inventory" "this" {}
+
+resource "imagetest_harness_docker" "test" {
+  name = "test"
+  inventory = data.imagetest_inventory.this
+}
+
+resource "imagetest_feature" "test" {
+  name = "Simple Docker based test"
+  description = "Verify that docker images runs"
+  harness = imagetest_harness_docker.test
+  steps = [
+    {
+      name = "Echo"
+      cmd = "docker images"
```

Comment:
```suggestion
      cmd = "docker run hello-world"
```

I think there's some value in actually testing the host socket is usable for something other than listing